### PR TITLE
fix(state-viewer): compare V1 and V2 ChunkExtras correctly in apply cmd

### PR DESCRIPTION
**Pull request by marcelo-gonzalez**
we don't want to blindly check equality because very old chunk extras will be stored as ChunkExtraV1, and the equality check will fail. So explicitly check that the validator stake iters (which is the only changed field between the two) give the same results

### CODE REVIEW

File: tools/state-viewer/src/commands.rs [hunk 1]
```diff
@@ -415,6 +416,28 @@ pub(crate) fn get_receipt(receipt_id: CryptoHash, near_config: NearConfig, store
...
+    if l.gas_limit() != r.gas_limit() {
+        return false;
+    }
+    if l.balance_burnt() != r.balance_burnt() {
+        return false;
+    }
+    l.validator_proposals().zip_longest(r.validator_proposals()).all(|x| match x {
+        EitherOrBoth::Both(l, r) => l == r,
+        _ => false,
+    })
```
Comment (wacban) [inline tools/state-viewer/src/commands.rs:438]: Would the following work? It's a bit easier to read. 

```
if l.validator_proposals().collect() != r.validator_proposals().collect() {
  return false;
}
return true;
```

**Review by wacban — COMMENTED** [review]
Looks good, the only worry I have is that this code will get outdated if another ChunkExtra version gets added. Would it make sense to move the equality method to where ChunkExtras are defined? Or maybe write a check that will not compile if another version is added like and assert or a match e.g.
```
fn check_supported(chunk_extra: ChunkExtra) -> Error {
  match chunk_extra {
    V1(_) => Ok(()),
    V2(_) => Ok(()),
    // no default so that a new version would pop up as compile error
  } 
}
```

**Comment by marcelo-gonzalez** [comment]
> Looks good, the only worry I have is that this code will get outdated if another ChunkExtra version gets added. Would it make sense to move the equality method to where ChunkExtras are defined? Or maybe write a check that will not compile if another version is added like and assert or a match e.g.
> 
> ```
> fn check_supported(chunk_extra: ChunkExtra) -> Error {
>   match chunk_extra {
>     V1(_) => Ok(()),
>     V2(_) => Ok(()),
>     // no default so that a new version would pop up as compile error
>   } 
> }
> ```

yea good point. added a match at the beginning that wont compile if a V3 gets added

**Review by wacban — APPROVED** [review]
You, sir, have my approval.

**Pull request merged**